Doc/reference/datamodel.rst

```diff
@@ -1716,7 +1716,7 @@ the descriptor defines :meth:`__set__` and/or :meth:`__delete__`, it is a data
 descriptor; if it defines neither, it is a non-data descriptor.  Normally, data
 descriptors define both :meth:`__get__` and :meth:`__set__`, while non-data
 descriptors have just the :meth:`__get__` method.  Data descriptors with
-:meth:`__set__` and :meth:`__get__` defined always override a redefinition in an
+:meth:`__get__` defined always override a redefinition in an
```

Comment:
The new wording is confusing it that relies heavily on a reader fully understanding that "data descriptor" means that ``__set__`` or ``__delete__`` is defined; otherwise, the new wording can easily be misread as contradicting the sentence that came before it.

The current wording is clear and covers the common case.

If you really feel like the current wording must change because it doesn't cover the highly usual case of ``__delete__`` being defined but not ``__set__``, then you can just modify the original as to:

'''
    :meth:`__set__` (and/or :meth:`__delete__`) and :meth:`__get__` defined always override a redefinition ...
'''


But do consider that a proposed modification while technically correct will make the overall paragraph harder to understand for someone didn't already having an understanding of data descriptors and what their significance.

Reply:
Thanks @rhettinger, I see your point. I have updated the PR with your suggestion (as for me the paragraph was confusing because `__delete__` was not mentioned).